From e585f7d97bb2fdb54656fd894297578e2a0c94d1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Tue, 20 Aug 2019 02:36:36 +0200
Subject: [PATCH 107/152] scripting: Disable an eslint warning

Calling await in a loop means the asynchronous operations are
run sequentially instead of in-parallel. Usually that's not
what's wanted, so eslint has a rule to warn about this.

However here we use async/await to handle control back to the
mainloop between steps, so running operations sequentially is
actually intended.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/731
---
 js/ui/scripting.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/scripting.js b/js/ui/scripting.js
index 7a14fcf55..84c1ad183 100644
--- a/js/ui/scripting.js
+++ b/js/ui/scripting.js
@@ -317,7 +317,7 @@ async function runPerfScript(scriptModule, outputFile) {
 
     for (let step of scriptModule.run()) {
         try {
-            await step;
+            await step; // eslint-disable-line no-await-in-loop
         } catch (err) {
             log(`Script failed: ${err}\n${err.stack}`);
             Meta.exit(Meta.ExitCode.ERROR);
-- 
2.23.0

